docs/spec/spec.md

```diff
@@ -0,0 +1,380 @@
+        }
+    }
+);
+```
+
+### 2.2 Hub Service
+
+`websubhub:Service` is responsible to handle the received events. Underlying `http:Service` will receive the original 
+request, and then it will trigger the WebSubHub dispatcher which will invoke the respective remote method with the event 
+details.
+
+Following is the type-definition for `websubhub:Service`.
+```ballerina
+public type Service distinct service object {
+    // Sample POST request hub.mode=register&hub.topic=http://foo.com/bar
+    // Sample 200 OK response hub.mode=accepted or 200 OK hub.mode=denied&hub.reason=unauthorized
+    remote function onRegisterTopic(websubhub:TopicRegistration msg)
+        returns websubhub:TopicRegistrationSuccess|websubhub:TopicRegistrationError|error;
+
+    // Sample POST request hub.mode=unregister&hub.topic=http://foo.com/bar
+    // Sample 200 OK response hub.mode=accepted or 200 OK hub.mode=denied&hub.reason=unauthorized 
+    remote function onDeregisterTopic(websubhub:TopicDeregistration msg)
+        returns websubhub:TopicDeregistrationSuccess|websubhub:TopicDeregistrationError|error;
+
+    // Sample POST request with content type x-www-form-urlencoded hub.mode=publish&hub.topic=http://foo.com/bar
+    // for other content types such as xml, json and octect-stream hub.mode=publish should be in query string.
+    // Sample 200 OK response hub.mode=accepted or 200 OK hub.mode=denied&hub.reason=unauthorized 
+    remote function onUpdateMessage(websubhub:UpdateMessage msg)
+        returns websubhub:Acknowledgement|websubhub:UpdateMessageError|error;
+
+    // Sample POST request hub.mode=subscribe&hub.topic=http://foo.com/bar 
+    remote function onSubscription(websubhub:Subscription msg)
+        returns websubhub:SubscriptionAccepted|websubhub:SubscriptionPermanentRedirect|
+        websubhub:SubscriptionTemporaryRedirect|websubhub:BadSubscriptionError|
+        websubhub:InternalSubscriptionError|error;
+        
+    remote function onSubscriptionValidation(websubhub:Subscription msg)
+        returns websubhub:SubscriptionDeniedError|error?;
+
+    remote function onSubscriptionIntentVerified(websubhub:VerifiedSubscription msg) returns error?;
+
+    // Sample POST request hub.mode=unsubscribe&hub.topic=http://foo.com/bar
+    remote function onUnsubscription(websubhub:Unsubscription msg)
+        returns websubhub:UnsubscriptionAccepted|websubhub:BadUnsubscriptionError|
+        websubhub:InternalUnsubscriptionError|error;
+
+    remote function onUnsubscriptionValidation(websubhub:Unsubscription msg)
+        returns websubhub:UnsubscriptionDeniedError|error?;
+
+    remote function onUnsubscriptionIntentVerified(websubhub:VerifiedUnsubscription msg) returns error?;
+};
+```
+
+While the below remote methods are strictly WebSub compliant,
+- onSubscription 
+- onSubscriptionValidation
+- onSubscriptionIntentVerified 
+- onUnsubscritpion 
+- onUnsubscriptionValidation
+- onUnsubscriptionIntenVerified
```

Comment:
I think we should briefly explain the functionally of each remote method.

Reply:
Fixed with [19f7114601a4bb48d08efdf5331d55779437e8bb](https://github.com/ballerina-platform/module-ballerina-websubhub/pull/378/commits/19f7114601a4bb48d08efdf5331d55779437e8bb)